test/unit-test/cellular_common_api_utest.c

```diff
@@ -1024,3 +1048,76 @@ void test_Cellular_CommonSocketRegisterClosedCallback_Happy_Path( void )
     TEST_ASSERT_EQUAL( socketHandle.closedCallback, cellularSocketClosedCallback );
     TEST_ASSERT_EQUAL( socketHandle.pClosedCallbackContext, testCallback );
 }
+
+/**
+ * @brief Test that option set local port happy path case for Cellular_CommonSocketSetSockOpt.
+ */
+void test_Cellular_CommonSocketSetSockOpt_Option_SetLocalPort_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    memset( &context, 0, sizeof( CellularContext_t ) );
+    struct CellularSocketContext socketHandle;
+    uint16_t optionValue = 12345;
+
+    socketHandle.socketState = SOCKETSTATE_ALLOCATED;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonSocketSetSockOpt( &context, &socketHandle,
+                                                      CELLULAR_SOCKET_OPTION_LEVEL_TRANSPORT,
+                                                      CELLULAR_SOCKET_OPTION_SET_LOCAL_PORT,
+                                                      ( const uint8_t * ) &optionValue, sizeof( uint16_t ) );
+
+    TEST_ASSERT_EQUAL( optionValue, socketHandle.localPort );
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test that option set local port at wrong socket state for Cellular_CommonSocketSetSockOpt.
+ */
+void test_Cellular_CommonSocketSetSockOpt_Option_SetLocalPort_Failure_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    memset( &context, 0, sizeof( CellularContext_t ) );
+    struct CellularSocketContext socketHandle;
+    uint16_t optionValue = 12345;
+
+    socketHandle.socketState = SOCKETSTATE_CONNECTING;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonSocketSetSockOpt( &context, &socketHandle,
+                                                      CELLULAR_SOCKET_OPTION_LEVEL_TRANSPORT,
+                                                      CELLULAR_SOCKET_OPTION_SET_LOCAL_PORT,
+                                                      ( const uint8_t * ) &optionValue, sizeof( uint16_t ) );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INTERNAL_FAILURE, cellularStatus );
```

Comment:
Add an assert that `socketHandle.localPort` did not change.

Reply:
Added. Thanks!